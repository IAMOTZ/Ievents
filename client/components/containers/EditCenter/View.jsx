import React from 'react';
import PropTypes from 'prop-types';
import SideNavigation from '../../common/SideNavigation';
import Header from '../../common/Header';
import { AuthTopNavigation } from '../../common/TopNavigation';
import CenterForm from '../AddCenter/CenterForm';

const View = props => (
  <div className="add-center-container">
    <AuthTopNavigation
      name={props.userName}
      title="Edit Center"
      isAdmin={props.isAdmin}
      isSuperAdmin={props.isSuperAdmin}
      dispatch={props.dispatch}
    />
    <div className="container-fluid">
      <div className="row">
        <SideNavigation
          name={props.userName}
          isAdmin={props.isAdmin}
          isSuperAdmin={props.isSuperAdmin}
          dispatch={props.dispatch}
        />
        <div className="col-lg-10 offset-md-2">
          <Header text="Edit Center" />
          <div className="page-content">
            <CenterForm
              updatingCenterStarted={props.updatingCenterStarted}
              updatingCenterError={props.updatingCenterError}
              getInput={props.getInput}
              inputErrors={props.inputErrors}
              handleImageDrop={props.handleImageDrop}
              images={props.images}
              update={props.update}
              toUpdate={props.toUpdate}
            />
          </div>
        </div>
      </div>
    </div>
    <footer className="d-block d-sm-none mt-5">
      <div className="container text-white text-center py-5">
        <h1>iEvents</h1>
        <p>Copyright &copy; 2017</p>
      </div>
    </footer>
  </div>
);

View.defaultProps = {
  updatingCenterError: '',
  inputErrors: {},
  images: [],
};

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  userName: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  updatingCenterStarted: PropTypes.bool.isRequired,
  updatingCenterError: PropTypes.string,
  getInput: PropTypes.func.isRequired,
  inputErrors: PropTypes.object,
  handleImageDrop: PropTypes.func.isRequired,
  images: PropTypes.array,
  update: PropTypes.func.isRequired,
  toUpdate: PropTypes.object.isRequired,
};

export default View;
