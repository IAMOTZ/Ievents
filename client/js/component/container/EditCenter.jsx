import React from 'react';
import Dropzone from 'react-dropzone';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import { updateCenter, clearStatus } from '../../actions/centerActions';

import UserSideNav from '../common/SideNavigation.jsx';
import Header from '../common/Header.jsx';
import ImageInput from '../common/ImageInput.jsx';
import { UserTopNav } from '../common/TopNavigation.jsx';
import { WarningAlert } from '../common/Alert';

@connect((store) => {
  return {
    user: store.user.user,
    authenticated: store.user.status.fetched,
    toEdit: store.centers.toEdit,
    status: {
      error: store.centers.status.updatingError.message,
      success: store.centers.status.updated,
    }
  }
})

export default class EditCenter extends React.Component {
  constructor() {
    super();
    this.state = {
      name: null,
      location: null,
      details: null,
      capacity: null,
      price: null,
      newImages: null,
    }
  }

  componentWillUnmount() {
    this.props.dispatch(clearStatus('ALL'));
  }

  // This method uses user input to update the state
  getInput = (e) => {
    const state = this.state;
    state[e.target.name] = e.target.value;
    this.setState(state);
  }

  handleImageDrop = (files) => {
    this.setState({
      newImages: files,
    });
  }

  // This method fires the action to create an event
  update = () => {
    const {
      name,
      location,
      details,
      capacity,
      price,
      newImages,
    } = this.state;
    const centerId = this.props.toEdit.id;
    const centerDetails = { name, location, details, capacity, price, };
    const fd = new FormData();
    for (let detail in centerDetails) {
      if (centerDetails[detail]) {
        fd.append(`${detail}`, centerDetails[detail]);
      }
    }
    if (newImages) {
      for (let i = 0; i < newImages.length; i += 1) {
        fd.append('images', newImages[i]);
      }
    }
    this.props.dispatch(updateCenter(centerId, fd, this.props.user.token));
  }

  render() {
    if (!this.props.authenticated) {
      return (<Redirect to="/users/login" />)
    } else if (this.props.status.success) {
      return (<Redirect to="/centers2" />);
    } else {
      return (
        <div className="add-center-container">
          {/* Top navigation on small screen */}
          <UserTopNav name={this.props.user.name} title='Edit Center' />

          <div class="container-fluid">
            <div class="row">

              {/*  Side navigation on large screen */}
              <UserSideNav userName={this.props.user.name} />

              {/* Main content */}
              <div class="col-lg-10 offset-md-2" id="add-event-section">

                {/* Content Header(navigation) on large screen */}
                <Header text='Edit Center' />

                {/* Input form */}
                <form class="mt-lg-5 mb-md-5">
                  <div className="w-lg-50">
                    <WarningAlert message={this.props.status.error} />
                  </div>
                  <div class="form-group row">
                    <label for="name" class="col-sm-1 col-form-label">Name</label>
                    <div class="col-sm-11">
                      <input type="email"
                        defaultValue={this.props.toEdit.name}
                        class="form-control w-lg-50"
                        id="name"
                        name="name"
                        placeholder="The name of the center"
                        onChange={this.getInput} />
                      <small class="form-text text-muted">Less than 30 characters</small>
                    </div>
                  </div>
                  <div class="form-group row">
                    <label for="location" class="col-sm-1 col-form-label">Location</label>
                    <div class="col-sm-11">
                      <input type="text"
                        defaultValue={this.props.toEdit.location}
                        class="form-control w-lg-50"
                        id="location"
                        name="location"
                        placeholder="The location of the center"
                        onChange={this.getInput} />
                      <small class="form-text text-muted">Less than 30 characters</small>
                    </div>
                  </div>
                  <div class="form-group row">
                    <label for="details" class="col-sm-1 col-form-label">Details</label>
                    <div class="col-sm-11">
                      <textarea
                        defaultValue={this.props.toEdit.details}
                        class="form-control w-lg-50"
                        id="datails" rows="7"
                        name="details"
                        placeholder="More details about the center"
                        onChange={this.getInput}></textarea>
                      <small class="form-text text-muted">Less than 200 characters</small>
                    </div>
                  </div>
                  <div class="form-group row">
                    <label for="capacity" class="col-sm-1 col-form-label">Capacity</label>
                    <div class="col-sm-11">
                      <input type="number"
                        defaultValue={this.props.toEdit.capacity}
                        class="form-control w-lg-50"
                        id="capacity"
                        name="capacity"
                        placeholder="How many seats"
                        onChange={this.getInput} />
                    </div>
                  </div>
                  <div class="form-group row">
                    <label for="price" class="col-sm-1 col-form-label">Price</label>
                    <div class="col-sm-11">
                      <input type="number"
                        defaultValue={this.props.toEdit.price}
                        class="form-control w-lg-50"
                        id="price"
                        name="price"
                        placeholder="Price"
                        onChange={this.getInput} />
                    </div>
                  </div>
                  <div class="form-group">
                    <label for="image">Image</label>
                    <div className="text-center">
                      <ImageInput
                        id="image"
                        onDrop={this.handleImageDrop}
                        newImage={this.state.newImages ? this.state.newImages[0] : null}
                        previousImage={this.props.toEdit.images ? this.props.toEdit.images[0] : null}
                      />
                    </div>
                  </div>
                  <div class="text-center w-25 pt-3">
                    <a class="btn btn-outline-dark" onClick={this.update}>Update</a>
                  </div>
                </form>

              </div>
            </div>
          </div>

          <footer class="d-block d-sm-none mt-5">
            <div class="container text-white text-center py-5">
              <h1>Ievents</h1>
              <p>Copyright &copy; 2017</p>
            </div>
          </footer>
        </div>
      );
    }
  }
}
