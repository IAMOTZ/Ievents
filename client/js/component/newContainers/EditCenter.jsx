import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { find } from 'lodash';
import { validateUpdateCenterInputs } from '../../helpers/inputValidators';
import { updateCenter } from '../../actions/centerActions';
import { stopAsyncProcess } from '../../actions/commonActions';
import * as asyncProcess from '../../actions/asyncProcess';
import UserSideNav from '../common/SideNavigation';
import Header from '../common/Header';
import ImageInput from '../common/ImageInput';
import { UserTopNav } from '../common/TopNavigation';
import { LoadingIcon } from '../common/LoadingAnimation';
import { BigAlert, SmallAlert } from '../common/Alert';

@connect((store) => {
  const { user } = store.authReducer;
  const { centers } = store.fetchCentersReducer;
  return {
    user,
    isAdmin: user.role === 'admin',
    isSuperAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isUserAuthenticaed: store.authReducer.loggingUserResolved,
    toUpdate: find(centers, { id: store.updateCenterReducer.centerToUpdate }),
    updatingCenterStarted: store.updateCenterReducer.updatingCenterStarted,
    updatingCenterResolved: store.updateCenterReducer.updatingCenterResolved,
    updatingCenterError: store.updateCenterReducer.updatingCenterError,
  };
})
class EditCenter extends React.Component {
  constructor() {
    super();
    this.state = {
      name: null,
      location: null,
      details: null,
      capacity: null,
      price: null,
      newImages: null,
      inputErrors: {
        nameError: null,
        locationError: null,
        detailsError: null,
        capacityError: null,
        priceError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.UPDATING_CENTER));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      nameError: null,
      locationError: null,
      detailsError: null,
      capacityError: null,
      priceError: null,
    };
    this.setState(state);
  }

  /**
   * A callback to update the state after the user uploads an image.
   * @param {File} files The image file.
   */
  handleImageDrop = (files) => {
    this.setState({
      newImages: files,
    });
  }

  /**
   * Dispatches the action to update the center.
   * @param {Event} event The event object.
   */
  update = (event) => {
    event.preventDefault();
    this.props.dispatch(stopAsyncProcess(asyncProcess.UPDATING_CENTER));
    const {
      name, location, details, capacity, price, newImages,
    } = this.state;
    const centerId = this.props.toUpdate.id;
    const centerDetails = {
      name, location, details, capacity, price,
    };
    const inputErrors = validateUpdateCenterInputs(centerDetails);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      const fd = new FormData();
      const entries = Object.entries(centerDetails);
      entries.forEach((entry) => {
        const key = entry[0];
        const value = entry[1];
        if (value) {
          fd.append(key, value);
        }
      });
      if (newImages) {
        fd.append('image', newImages[0]);
      }
      this.props.dispatch(updateCenter(centerId, fd, this.props.user.token));
    }
    window.scrollTo(0, 0);
  }

  render() {
    if (!this.props.isUserAuthenticaed) {
      return (<Redirect to="/users/login" />);
    } else if (this.props.updatingCenterResolved) {
      return (<Redirect to="/centers2" />);
    }
    return (
      <div className="add-center-container">
        {/* Top navigation on small screen */}
        <UserTopNav
          name={this.props.user.name}
          title="Edit Center"
          isAdmin={this.props.isAdmin}
          isSuperAdmin={this.props.isSuperAdmin}
          dispatch={this.props.dispatch}
        />
        <div className="container-fluid">
          <div className="row">
            {/*  Side navigation on large screen */}
            <UserSideNav
              name={this.props.user.name}
              isAdmin={this.props.isAdmin}
              isSuperAdmin={this.props.isSuperAdmin}
              dispatch={this.props.dispatch}
            />
            {/* Main content */}
            <div className="col-lg-10 offset-md-2" id="add-event-section">
              {/* Content Header(navigation) on large screen */}
              <Header text="Edit Center" />
              {/* Input form */}
              <form className="mt-lg-5 mb-md-5">
                <LoadingIcon start={this.props.updatingCenterStarted} size={3} />
                <div className="w-lg-50 mx-auto">
                  <BigAlert message={this.props.updatingCenterError} />
                </div>
                <div className="container">
                  <div className="row">
                    <div className="col-12 col-lg-6">
                      <div className="form-group">
                        <label htmlFor="name">Name</label>
                        <input
                          type="text"
                          className="form-control"
                          id="name"
                          name="name"
                          placeholder="The name of the center"
                          defaultValue={this.props.toUpdate.name}
                          onChange={this.getInput}
                        />
                        <small className="form-text text-muted">Between 5 and 30 characters</small>
                        <SmallAlert message={this.state.inputErrors.nameError} />
                      </div>
                    </div>
                    <div className="col-12 col-lg-6">
                      <div className="form-group">
                        <label htmlFor="location">Location</label>
                        <input
                          type="text"
                          className="form-control"
                          id="location"
                          name="location"
                          placeholder="The location of the center"
                          defaultValue={this.props.toUpdate.location}
                          onChange={this.getInput}
                        />
                        <small className="form-text text-muted">Less than 50 characters</small>
                        <SmallAlert message={this.state.inputErrors.locationError} />
                      </div>
                    </div>
                  </div>
                  <div className="row">
                    <div className="col-12 col-lg-6">
                      <div className="form-group">
                        <label htmlFor="details">Details</label>
                        <textarea
                          className="form-control"
                          id="details"
                          rows="7"
                          name="details"
                          placeholder="More details about the center"
                          defaultValue={this.props.toUpdate.details}
                          onChange={this.getInput}
                        />
                        <small className="form-text text-muted">Less than 300 characters</small>
                        <SmallAlert message={this.state.inputErrors.detailsError} />
                      </div>
                    </div>
                    <div className="col-12 col-lg-6">
                      <div className="row">
                        <div className="col-12">
                          <div className="form-group">
                            <label htmlFor="capacity">Capacity</label>
                            <input
                              type="number"
                              className="form-control"
                              id="capacity"
                              name="capacity"
                              placeholder="How many seats"
                              defaultValue={this.props.toUpdate.capacity}
                              onChange={this.getInput}
                            />
                            <SmallAlert message={this.state.inputErrors.capacityError} />
                          </div>
                        </div>
                      </div>
                      <div className="row">
                        <div className="col-12">
                          <div className="form-group">
                            <label htmlFor="price">Price</label>
                            <input
                              type="number"
                              className="form-control"
                              id="price"
                              name="price"
                              placeholder="Price"
                              defaultValue={this.props.toUpdate.price}
                              onChange={this.getInput}
                            />
                            <SmallAlert message={this.state.inputErrors.priceError} />
                          </div>
                        </div>
                      </div>
                      <div className="row">
                        <div className="col-12">
                          <div className="form-group">
                            <label htmlFor="image">Image</label>
                            <div className="text-center">
                              <ImageInput
                                style={{ height: '100px' }}
                                id="image"
                                onDrop={this.handleImageDrop}
                                newImage={this.state.newImages ? this.state.newImages[0] : null}
                                previousImage={
                                  this.props.toUpdate.images ? this.props.toUpdate.images[0] : null
                                }
                              />
                            </div>
                          </div>
                        </div>
                        <div className="ml-3 pt-3">
                          <button
                            id="update-btn"
                            className="btn btn-outline-dark"
                            disabled={this.props.updatingCenterStarted}
                            onClick={this.update}
                          >Update
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <footer className="d-block d-sm-none mt-5">
          <div className="container text-white text-center py-5">
            <h1>Ievents</h1>
            <p>Copyright &copy; 2017</p>
          </div>
        </footer>
      </div>
    );
  }
}

export default EditCenter;
