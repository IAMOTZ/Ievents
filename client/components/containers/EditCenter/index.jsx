import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { find } from 'lodash';
import { validateUpdateCenterInputs } from '../../../helpers/inputValidators';
import { updateCenter } from '../../../actions/centerActions';
import { stopAsyncProcess } from '../../../actions/commonActions';
import * as asyncProcess from '../../../actions/asyncProcess';
import View from './View';


@connect((store) => {
  const { user } = store.authReducer;
  const { centers } = store.fetchCentersReducer;
  return {
    userName: user.name,
    userToken: user.token,
    isAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isSuperAdmin: user.role === 'superAdmin',
    toUpdate: find(centers, { id: store.updateCenterReducer.centerToUpdate }),
    updatingCenterStarted: store.updateCenterReducer.updatingCenterStarted,
    updatingCenterResolved: store.updateCenterReducer.updatingCenterResolved,
    updatingCenterError: store.updateCenterReducer.updatingCenterError,
  };
})
class EditCenter extends React.Component {
  constructor() {
    super();
    this.state = {
      name: null,
      location: null,
      details: null,
      capacity: null,
      price: null,
      newImages: null,
      inputErrors: {
        nameError: null,
        locationError: null,
        detailsError: null,
        capacityError: null,
        priceError: null,
      },
    };
  }

  componentWillUnmount() {
    this.props.dispatch(stopAsyncProcess(asyncProcess.UPDATING_CENTER));
  }

  /**
   * Stores the user inputs in the state of this component.
   * @param {Event} event The event object.
   */
  getInput = (event) => {
    const state = { ...this.state };
    state[event.target.name] = event.target.value;
    this.setState(state);
  }

  /**
   * Clears all the inputErrors in the state.
   */
  clearInputErrors = () => {
    const state = { ...this.state };
    state.inputErrors = {
      nameError: null,
      locationError: null,
      detailsError: null,
      capacityError: null,
      priceError: null,
    };
    this.setState(state);
  }

  /**
   * A callback to update the state after the user uploads an image.
   * @param {File} files The image file.
   */
  handleImageDrop = (files) => {
    this.setState({
      newImages: files,
    });
  }

  /**
   * Dispatches the action to update the center.
   * @param {Event} event The event object.
   */
  update = (event) => {
    event.preventDefault();
    this.props.dispatch(stopAsyncProcess(asyncProcess.UPDATING_CENTER));
    const {
      name, location, details, capacity, price, newImages,
    } = this.state;
    const centerId = this.props.toUpdate.id;
    const centerDetails = {
      name, location, details, capacity, price,
    };
    const inputErrors = validateUpdateCenterInputs(centerDetails);
    if (inputErrors.errorFound) {
      const state = { ...this.state };
      state.inputErrors = inputErrors;
      this.setState(state);
    } else {
      const fd = new FormData();
      const entries = Object.entries(centerDetails);
      entries.forEach((entry) => {
        const key = entry[0];
        const value = entry[1];
        if (value) {
          fd.append(key, value);
        }
      });
      if (newImages) {
        fd.append('image', newImages[0]);
      }
      this.props.dispatch(updateCenter(centerId, fd, this.props.userToken));
    }
    window.scrollTo(0, 0);
  }

  render() {
    if (this.props.updatingCenterResolved) {
      return (<Redirect to="/centers" />);
    }
    return (
      <View
        userName={this.props.userName}
        isAdmin={this.props.isAdmin}
        isSuperAdmin={this.props.isSuperAdmin}
        dispatch={this.props.dispatch}
        updatingCenterStarted={this.props.updatingCenterStarted}
        updatingCenterError={this.props.updatingCenterError}
        getInput={this.getInput}
        inputErrors={this.state.inputErrors}
        handleImageDrop={this.handleImageDrop}
        images={this.state.images}
        update={this.update}
        toUpdate={this.props.toUpdate}
      />
    );
  }
}

EditCenter.defaultProps = {
  userName: '',
  userToken: '',
  isAdmin: false,
  isSuperAdmin: false,
  toUpdate: {},
  updatingCenterStarted: false,
  updatingCenterResolved: false,
  updatingCenterError: '',
  dispatch: () => {},
};

/* eslint-disable react/forbid-prop-types */
EditCenter.propTypes = {
  userName: PropTypes.string,
  userToken: PropTypes.string,
  isAdmin: PropTypes.bool,
  isSuperAdmin: PropTypes.bool,
  toUpdate: PropTypes.object,
  updatingCenterStarted: PropTypes.bool,
  updatingCenterResolved: PropTypes.bool,
  updatingCenterError: PropTypes.string,
  dispatch: PropTypes.func,
};

export default EditCenter;
