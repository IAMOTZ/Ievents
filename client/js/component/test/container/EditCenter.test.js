/* global shallow */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import EditCenter from '../../container/EditCenter.jsx';

const props = {
  user: { name: 'test-user' },
  authenticated: true,
  isAdmin: true,
  isSuperAdmin: false,
  toEdit: {
    name: 'test-center',
    location: 'test-location',
    details: 'test-details',
    capacity: 200,
    price: 100,
    images: null,
  },
  status: {
    updating: false,
    success: false,
    error: false,
  },
  dispatch: () => { },
};

const alterProps = newProps => ({
  user: newProps.user || props.user,
  authenticated: newProps.authenticated === undefined ? true : newProps.authenticated,
  isAdmin: newProps.isAdmin === undefined ? true : newProps.isAdmin,
  isSuperAdmin: newProps.isSuperAdmin || false,
  toEdit: {
    ...props.toEdit,
    ...newProps.toEdit,
  },
  status: {
    updating: newProps.updating || false,
    success: newProps.success || false,
    error: newProps.error || false,
  },
  dispatch: () => { },
});

describe('<EditCenter />', () => {
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<EditCenter {...props} />);
    })
    describe('Rendering:', () => {
      it('should render correctly on page load', () => {
        expect(wrapper).toMatchSnapshot();
      });
      it('should redirect to login page if user is not authenticated', () => {
        wrapper.setProps(alterProps({ authenticated: false }));
        expect(wrapper).toMatchSnapshot();
      });
      it('should redirect to centers page if center is successfully updated', () => {
        wrapper.setProps(alterProps({ success: true }));
        expect(wrapper).toMatchSnapshot();
      });
      it('should render correcty when updating', () => {
        wrapper.setProps(alterProps({ updating: true }));
        expect(wrapper).toMatchSnapshot();
      });
      it('should render correctly when there is an error', () => {
        wrapper.setProps(alterProps({ error: 'there was an error' }));
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('Behaviour:', () => {
      let wrapper;
      beforeEach(() => {
        wrapper = shallow(<EditCenter {...props} />);
      });
      it('should update the state when the user keys in the center name', () => {
        const name = 'test-center-updated';
        wrapper
          .find('#name')
          .simulate('change', { target: { name: 'name', value: name } });
        expect(wrapper.state('name')).toEqual(name);
      });
      it('should update the state when the user keys in the center location', () => {
        const location = 'test-location-updated';
        wrapper
          .find('#location')
          .simulate('change', { target: { name: 'location', value: location } });
        expect(wrapper.state('location')).toEqual(location);
      });
      it('should update the state when the user keys in the center details', () => {
        const details = 'test-details-updated';
        wrapper
          .find('#details')
          .simulate('change', { target: { name: 'details', value: details } });
        expect(wrapper.state('details')).toEqual(details);
      });
      it('should update the state when the user keys in the center capacity', () => {
        const capacity = 50;
        wrapper
          .find('#capacity')
          .simulate('change', { target: { name: 'capacity', value: capacity } });
        expect(wrapper.state('capacity')).toEqual(capacity);
      });
      it('should update the state when the user keys in the center price', () => {
        const price = 40;
        wrapper
          .find('#price')
          .simulate('change', { target: { name: 'price', value: price } });
        expect(wrapper.state('price')).toEqual(price);
      });
    });
  });
});
