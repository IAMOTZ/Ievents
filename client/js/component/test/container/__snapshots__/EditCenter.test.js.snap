// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditCenter /> Rendering: Rendering: should redirect to centers page if center is successfully updated 1`] = `
<Redirect
  push={false}
  to="/centers2"
/>
`;

exports[`<EditCenter /> Rendering: Rendering: should redirect to login page if user is not authenticated 1`] = `
<Redirect
  push={false}
  to="/users/login"
/>
`;

exports[`<EditCenter /> Rendering: Rendering: should render correctly on page load 1`] = `
<div
  className="add-center-container"
>
  <UserTopNav
    dispatch={[Function]}
    isAdmin={true}
    isSuperAdmin={false}
    name="test-user"
    title="Edit Center"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <UserSideNav
        dispatch={[Function]}
        isAdmin={true}
        isSuperAdmin={false}
        name="test-user"
      />
      <div
        className="col-lg-10 offset-md-2"
        id="add-event-section"
      >
        <Header
          text="Edit Center"
        />
        <form
          className="mt-lg-5 mb-md-5 w-lg-50"
        >
          <LoadingIcon
            size={2}
            start={false}
          />
          <WarningAlert
            message={false}
          />
          <div
            className="form-group"
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="form-control"
              defaultValue="test-center"
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="The name of the center"
              type="text"
            />
            <small
              className="form-text text-muted"
            >
              Less than 30 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="location"
            >
              Location
            </label>
            <input
              className="form-control"
              defaultValue="test-location"
              id="location"
              name="location"
              onChange={[Function]}
              placeholder="The location of the center"
              type="text"
            />
            <small
              className="form-text text-muted"
            >
              Less than 30 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="details"
            >
              Details
            </label>
            <textarea
              className="form-control"
              defaultValue="test-details"
              id="details"
              name="details"
              onChange={[Function]}
              placeholder="More details about the center"
              rows="7"
            />
            <small
              className="form-text text-muted"
            >
              Less than 200 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="capacity"
            >
              Capacity
            </label>
            <input
              className="form-control"
              defaultValue={200}
              id="capacity"
              name="capacity"
              onChange={[Function]}
              placeholder="How many seats"
              type="number"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="price"
            >
              Price
            </label>
            <input
              className="form-control"
              defaultValue={100}
              id="price"
              name="price"
              onChange={[Function]}
              placeholder="Price"
              type="number"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="image"
            >
              Image
            </label>
            <div
              className="text-center"
            >
              <ImageInput
                id="image"
                newImage={null}
                onDrop={[Function]}
                previousImage={null}
              />
            </div>
          </div>
          <div
            className="ml-3 pt-3"
          >
            <button
              className="btn btn-outline-dark"
              disabled={false}
              onClick={[Function]}
            >
              Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <footer
    className="d-block d-sm-none mt-5"
  >
    <div
      className="container text-white text-center py-5"
    >
      <h1>
        Ievents
      </h1>
      <p>
        Copyright © 2017
      </p>
    </div>
  </footer>
</div>
`;

exports[`<EditCenter /> Rendering: Rendering: should render correctly when there is an error 1`] = `
<div
  className="add-center-container"
>
  <UserTopNav
    dispatch={[Function]}
    isAdmin={true}
    isSuperAdmin={false}
    name="test-user"
    title="Edit Center"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <UserSideNav
        dispatch={[Function]}
        isAdmin={true}
        isSuperAdmin={false}
        name="test-user"
      />
      <div
        className="col-lg-10 offset-md-2"
        id="add-event-section"
      >
        <Header
          text="Edit Center"
        />
        <form
          className="mt-lg-5 mb-md-5 w-lg-50"
        >
          <LoadingIcon
            size={2}
            start={false}
          />
          <WarningAlert
            message="there was an error"
          />
          <div
            className="form-group"
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="form-control"
              defaultValue="test-center"
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="The name of the center"
              type="text"
            />
            <small
              className="form-text text-muted"
            >
              Less than 30 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="location"
            >
              Location
            </label>
            <input
              className="form-control"
              defaultValue="test-location"
              id="location"
              name="location"
              onChange={[Function]}
              placeholder="The location of the center"
              type="text"
            />
            <small
              className="form-text text-muted"
            >
              Less than 30 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="details"
            >
              Details
            </label>
            <textarea
              className="form-control"
              defaultValue="test-details"
              id="details"
              name="details"
              onChange={[Function]}
              placeholder="More details about the center"
              rows="7"
            />
            <small
              className="form-text text-muted"
            >
              Less than 200 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="capacity"
            >
              Capacity
            </label>
            <input
              className="form-control"
              defaultValue={200}
              id="capacity"
              name="capacity"
              onChange={[Function]}
              placeholder="How many seats"
              type="number"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="price"
            >
              Price
            </label>
            <input
              className="form-control"
              defaultValue={100}
              id="price"
              name="price"
              onChange={[Function]}
              placeholder="Price"
              type="number"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="image"
            >
              Image
            </label>
            <div
              className="text-center"
            >
              <ImageInput
                id="image"
                newImage={null}
                onDrop={[Function]}
                previousImage={null}
              />
            </div>
          </div>
          <div
            className="ml-3 pt-3"
          >
            <button
              className="btn btn-outline-dark"
              disabled={false}
              onClick={[Function]}
            >
              Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <footer
    className="d-block d-sm-none mt-5"
  >
    <div
      className="container text-white text-center py-5"
    >
      <h1>
        Ievents
      </h1>
      <p>
        Copyright © 2017
      </p>
    </div>
  </footer>
</div>
`;

exports[`<EditCenter /> Rendering: Rendering: should render correcty when updating 1`] = `
<div
  className="add-center-container"
>
  <UserTopNav
    dispatch={[Function]}
    isAdmin={true}
    isSuperAdmin={false}
    name="test-user"
    title="Edit Center"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <UserSideNav
        dispatch={[Function]}
        isAdmin={true}
        isSuperAdmin={false}
        name="test-user"
      />
      <div
        className="col-lg-10 offset-md-2"
        id="add-event-section"
      >
        <Header
          text="Edit Center"
        />
        <form
          className="mt-lg-5 mb-md-5 w-lg-50"
        >
          <LoadingIcon
            size={2}
            start={true}
          />
          <WarningAlert
            message={false}
          />
          <div
            className="form-group"
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="form-control"
              defaultValue="test-center"
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="The name of the center"
              type="text"
            />
            <small
              className="form-text text-muted"
            >
              Less than 30 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="location"
            >
              Location
            </label>
            <input
              className="form-control"
              defaultValue="test-location"
              id="location"
              name="location"
              onChange={[Function]}
              placeholder="The location of the center"
              type="text"
            />
            <small
              className="form-text text-muted"
            >
              Less than 30 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="details"
            >
              Details
            </label>
            <textarea
              className="form-control"
              defaultValue="test-details"
              id="details"
              name="details"
              onChange={[Function]}
              placeholder="More details about the center"
              rows="7"
            />
            <small
              className="form-text text-muted"
            >
              Less than 200 characters
            </small>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="capacity"
            >
              Capacity
            </label>
            <input
              className="form-control"
              defaultValue={200}
              id="capacity"
              name="capacity"
              onChange={[Function]}
              placeholder="How many seats"
              type="number"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="price"
            >
              Price
            </label>
            <input
              className="form-control"
              defaultValue={100}
              id="price"
              name="price"
              onChange={[Function]}
              placeholder="Price"
              type="number"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="image"
            >
              Image
            </label>
            <div
              className="text-center"
            >
              <ImageInput
                id="image"
                newImage={null}
                onDrop={[Function]}
                previousImage={null}
              />
            </div>
          </div>
          <div
            className="ml-3 pt-3"
          >
            <button
              className="btn btn-outline-dark"
              disabled={true}
              onClick={[Function]}
            >
              Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <footer
    className="d-block d-sm-none mt-5"
  >
    <div
      className="container text-white text-center py-5"
    >
      <h1>
        Ievents
      </h1>
      <p>
        Copyright © 2017
      </p>
    </div>
  </footer>
</div>
`;
